mailpoet/lib/Segments/DynamicSegments/Filters/WooCommerceUsedPaymentMethod.php

```diff
@@ -52,27 +53,30 @@ public function apply(QueryBuilder $queryBuilder, DynamicSegmentFilterEntity $fi
       throw new InvalidFilterException('Missing payment methods', InvalidFilterException::MISSING_VALUE);
     }
 
-    if (!is_int($days) || $days < 1) {
-      throw new InvalidFilterException('Missing days', InvalidFilterException::MISSING_VALUE);
+    $data = $filterData->getData();
+    // Backwards compatibility for filters saved before timeframe was added
+    if (!isset($data['timeframe'])) {
```

Comment:
Have you considered adding the `timeframe` in the migration instead of having this backward compatibility code? Maybe it is better to have it here in case this code runs before the migration is executed?

Reply:
Thanks for this suggestion! I think adding it to the migrationand removing the backwards compatibility code makes a lot more sense. I've updated the migration in 7b12046ceec7eaefa9ddb071a833398e70ca48b8. Can you think of any way that the new migration could potentially fail? The update is wrapped in some conditionals, but they should always be true for any saved filters.

Reply:
Not really, I just assumed that was why you opted to add the BC code. Creating the `timeframe` in the migration seems like a good approach to me.